src/libsyncengine/update_detection/update_detector/updatetreeworker.cpp

```diff
@@ -152,38 +171,25 @@ ExitCode UpdateTreeWorker::step3DeleteDirectory() {
                         << L"') updated. Operation DELETE inserted in change events.");
             }
         } else {
-            // Look for parentNodeId in db
-            std::optional<NodeId> parentNodeId;
-            bool found = false;
-            if (!_syncDb->id(_side, deleteOp->path().parent_path(), parentNodeId, found)) {
-                LOG_SYNCPAL_WARN(_logger, "Error in SyncDb::id");
-                return ExitCodeDbError;
-            }
-            std::shared_ptr<Node> parentNode = nullptr;
-            bool ok = false;
-            if (found && parentNodeId.has_value()) {
-                auto nodeIt = _updateTree->nodes().find(*parentNodeId);
-                if (nodeIt != _updateTree->nodes().end()) {
-                    parentNode = nodeIt->second;
-                    ok = true;
-                }
-            }
+            std::shared_ptr<Node> parentNode;
+            if (const auto searchExitCode = searchForParentNode(deleteOp->path(), parentNode); searchExitCode != ExitCodeOk) {
+                return searchExitCode;
+            };
 
-            if (!ok) {
+            if (!parentNode) {
                 SyncPath newPath;
-                ExitCode exitCode = getNewPathAfterMove(deleteOp->path(), newPath);
-                if (exitCode != ExitCodeOk) {
-                    if (exitCode == ExitCodeDbError) {
-                        LOG_SYNCPAL_WARN(_logger, "Error in UpdateTreeWorker::getNewPathAfterMove");
-                        return exitCode;
-                    }
+                if (const auto newPathExitCode = getNewPathAfterMove(deleteOp->path(), newPath);
+                    newPathExitCode == ExitCodeDbError) {
```

Comment:
This is not directly related to your PR but what happen if the 'newPathExitCode' is an error different from 'ExitCodeDbError' ('ExitCodeDataError' for exemple)?

Reply:
Thanks for noticing this. I am under the impression that it would be better to return immediately the error code if this code is not `ExitCodeOk.` This is what I implement with the next commit.